src/node_http_parser.cc

```diff
@@ -167,6 +168,23 @@ class Parser : public AsyncWrap, public StreamListener {
     num_fields_ = num_values_ = 0;
     url_.Reset();
     status_message_.Reset();
+
+    Local<Object> obj = object();
+    Local<Value> cb = obj->Get(env()->context(),
+                               kOnMessageBegin).ToLocalChecked();
+
+    if (!cb->IsFunction())
```

Comment:
will it affect performance? Not sure if there is a better way. An extra callback will be invoked for every request